src/main/kotlin/app/Main.kt

```diff
@@ -1,25 +1,59 @@
 // Copyright 2017 Sourcerer Inc. All Rights Reserved.
+// Author: Anatoly Kislov (anatoly@sourcerer.io)
 
 package app
 
-import app.utils.Args
+import app.utils.CommandConfig
+import app.utils.CommandExplore
+import app.utils.Options
 import com.beust.jcommander.JCommander
 import com.beust.jcommander.ParameterException
+import kotlin.system.exitProcess
+
+fun main(argv: Array<String>) {
+    val commandExplore = "explore"
+    val commandConfig = "config"
+
+    val options = Options()
+    val explore = CommandExplore()
+    val config = CommandConfig()
+    val jc: JCommander = JCommander.newBuilder()
+            .programName("sourcerer")  // Used for usage method.
+            .addObject(options)
+            .addCommand(commandExplore, explore)
+            .addCommand(commandConfig, config)
+            .build()
+
+    Configurator.setCurrentOptions(options)
 
-fun main(argv : Array<String>) {
-    val args = Args()
     try {
-        JCommander.newBuilder()
-                .addObject(args)
-                .build()
-                .parse(*argv)
+        jc.parse(*argv)
     } catch (e: ParameterException) {
         println(e.message)
-        return
+        exitProcess(1)  // Reporting failure with non-zero exit code.
+    }
+
+    when (jc.parsedCommand) {
+        commandExplore -> explore(explore)
+        commandConfig -> config(config)
+        else -> help(jc)  // Show help info if no command specified.
     }
 
-    println(arrayOf(args.username, args.password, args.path).joinToString())
+    println(arrayOf(options.username, options.password).joinToString())
+    println(jc.parsedCommand)
+    println(explore.path)
+}
+
+fun explore(commandOptions: CommandExplore) {
```

Comment:
Shouldn't it be just: add, remove, list? Is there anything else that may have the same set of commands?